stl/inc/execution

```diff
@@ -4430,20 +4474,36 @@ struct _Static_partitioned_exclusive_scan2 {
             return _Cancellation_status::_Running;
         }
 
-        _STD vector<_Ty> _Intermediate_result;
-
-        // Calculate local sum and publish to other threads
-        const auto _Last = _STD _Exclusive_scan_per_chunk(
-            _In_range._First, _In_range._Last, _Dest, _Reduce_op, _Chunk->_Local._Ref(), _Intermediate_result);
-        _Chunk->_Store_available_state(_Local_available);
+        // If the intermediate type is same with output type, then we can direct store the bop result in the output
+        if constexpr (_STD is_same_v<_Ty, typename std::iterator_traits<_FwdIt2>::value_type>) {

```

Comment:
We generally don't `_STD`- or `std::`-qualify types or variables unless needed. Ditto for other occurences of `is_same_v`, `iterator_traits` and `vector`.

It may be better to write `_Iter_value_t<_FwdIt2>` here, but see also #4436 and LWG-4080.

Reply:
_Iter_value_t is C++20 things, here isn't.

Reply:
> _Iter_value_t is C++20 things, here isn't.

`iter_value_t` is a C++20 stuff, while `_Iter_value_t` is not.

Reply:
What's the benefit to replace it?